hld/daemon/daemon.go

```diff
@@ -63,9 +65,16 @@ func New() (*Daemon, error) {
 		}
 	}
 
+	// Create session manager
+	sessionManager, err := session.NewManager()
```

Comment:
for testability etc, this should probably be 

1. an interface
2. dependency-injected with something like `dig` or even `fx` (but probably not fx)